### PR TITLE
This improves our support for clang under Windows.

### PR DESCRIPTION
**Pull request by lemire**
Fixes https://github.com/simdjson/simdjson/issues/763

### CODE REVIEW

**Comment by lemire** [comment]
@jkeiser  This PR uses @pps83's trick of forcing the inclusion of specific headers (which one is not supposed to do) and it therefore avoids the messiness of having to do things like `#define __AVX2__ 1` which is objectively worse. This approach should also be more robust.

Compared to @pps83's PR (and the current master), this PR carefully distinguishes between clang-under-windows and regular visual studio...

**Comment by lemire** [comment]
@jkeiser  I merged PR https://github.com/simdjson/simdjson/pull/818 into this...  so we will know, some time in the next few days (joke) whether my PR fixes the CI tests.

**Comment by lemire** [comment]
AH. Of all things, it is the new getopt thing that hits me.

**Comment by jkeiser** [comment]
@lemire is it? It looks like it has some kind of issue when it tries to set up the checkperf reference branch repository. I have a PR coming that will mitigate this ... we shouldn't be building the whole tree when we fuzz anyway.

**Comment by jkeiser** [comment]
@lemire #822 will likely get us past this (I haven't analyzed or tried to fix the underlying issue).

**Comment by jkeiser** [comment]
LOL I see, you are plagued with both problems.

**Comment by lemire** [comment]
@jkeiser Please review.

File: src/westmere/intrinsics.h [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
 
-#ifdef _MSC_VER
-#include <intrin.h> // visual studio
+#ifdef SIMDJSON_VISUAL_STUDIO
+// under clang within visual studio, this will include <x86intrin.h>
+#include <intrin.h> // visual studio or clang
 #else
 #include <x86intrin.h> // elsewhere
-#endif // _MSC_VER
+#endif // SIMDJSON_REGULAR_VISUAL_STUDIO
```
Comment (jkeiser) [inline src/westmere/intrinsics.h:9]: Non-matching comment

File: src/haswell/intrinsics.h [hunk 2]
```diff
@@ -3,17 +3,37 @@
...
+ * (e.g., arch:AVX2, -mavx2, etc.) which compiles the whole
+ * software with these advanced instructions. In simdjson, we
+ * want to compile the whole program for a generic target,
+ * and only target our specific kernels. As a workaround,
+ * we directly include the needed headers. These headers would
+ * normally guard against such usage, but we carefully included
+ * <x86intrin.h>  (or <intrin.h>) before, so the headers
+ * are fooled.
+ */
+#include <bmiintrin.h>   // for _blsr_u64
```
[the remark this answers is not in the document]
Comment (lemire) [reply src/haswell/intrinsics.h:32]: That's a good point. I was worried about this but forgot to check.

[the remark this answers is not in the document]
Comment (lemire) [reply src/haswell/intrinsics.h:32]: There is indeed a problem.

File: src/westmere/intrinsics.h (hunk 1 above)
Comment (lemire) [reply src/westmere/intrinsics.h]: Fixed in a subsequent commit.

File: src/haswell/intrinsics.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (lemire) [reply src/haswell/intrinsics.h:32]: Fixed in a commit.

File: include/simdjson/portability.h [hunk 3]
```diff
@@ -115,11 +142,18 @@ compiling for a known 64-bit platform."
 #define NO_SANITIZE_UNDEFINED
 #endif
 
-#ifdef _MSC_VER
+
+#ifdef SIMDJSON_VISUAL_STUDIO
+// We include intrin.h even if we are using clang under
+// visual studio. It is fine because clang comes with such
+// a header.
```
[the remark this answers is not in the document]
Comment (lemire) [reply include/simdjson/portability.h]: It is not needed. It is a leftover of older code. It did not cause harm, but I removed it.

**Comment by lemire** [comment]
I am waiting for this to complete its run. I think that this is quite cool.

**Comment by jkeiser** [comment]
This is indeed super exciting! It lets us build Windows executables with all the optimization we normally have :)

**Pull request merged**